### PR TITLE
add AMax F32/F16 feature

### PR DESCRIPTION
**Pull request by TonyYHsieh**
add AbsMax feature to hipBlasLt with type F32 and F16

### CODE REVIEW

**Review by KKyang — CHANGES_REQUESTED** [review]
Add to sample or use bench instead.

**Comment by TonyYHsieh** [comment]
> Add to sample or use bench instead.

remove sample first.
will develop ExtOp samples together.

**Comment by TonyYHsieh** [comment]
> Add to sample or use bench instead.

add AMax sample

**Comment by jichangjichang** [comment]
@TonyYHsieh  per our dicussion, the AMax's output scalar should support FP32.

**Comment by TonyYHsieh** [comment]
> @TonyYHsieh per our dicussion, the AMax's output scalar should support FP32.

add support FP32 output scale for FP16/FP32 input

File: tensilelite/Tensile/Ops/AMaxGenerator.py [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+        if (self.i_type.isHalf()):
+            mod.add(ti.VMaxF16(ti.vgpr("Output"), ti.vgpr("Output"), ti.vgpr("OutputB")))
+        elif (self.i_type.isSingle()):
+            mod.add(ti.VMaxF32(ti.vgpr("Output"), ti.vgpr("Output"), ti.vgpr("OutputB")))
+
+        return mod
+
+
+    def intra_wave_reduction(self) -> ti.Module:
+        label = ti.Label("permute", f'permuge')
```
Comment (Serge45) [inline tensilelite/Tensile/Ops/AMaxGenerator.py:460]: nitpick: typo here.

Comment (TonyYHsieh) [reply tensilelite/Tensile/Ops/AMaxGenerator.py]: fix typo

**Review by KKyang — APPROVED** [review]

**Review by Serge45 — APPROVED** [review]

**Pull request merged**